internal/command/ip/ip.go

```diff
@@ -40,12 +42,20 @@ func runIP(cmd *cobra.Command, args []string) error {
 		return err
 	}
 
-	vmDir, err := local.Open(localName)
+	// Open the VM directory and read its configuration under a global lock
+	vmConfig, err := globallock.With(func() (*vmconfig.VMConfig, error) {
```

Comment:
What do you think of moving the `globallocak` and partly `vmDir.FileLock()` login into the `Open` method to remove duplication. `local.Open` should itself always use a global lock and if we open it for "writing" then also do a file lock.

Reply:
I've tried this approach and it doesn't look very appealing: it results in duplicating the `Open()` methods (because not every call needs to lock the VM after opening) and makes the code harder to audit because the code that takes global locks is buried below.

For example, in `vetu delete`, we delete the VMs under a global lock. To do that, we call `local.Delete()`, for example. However, `Delete()` itself needs to lock the VM to ensure exclusivity. To do that, it calls the `OpenTryLocked()` method which tries to grab a global lock. This results in a deadlock.

Reply:
This works for `temporary` storage, however, because it has only two methods to interact with it, see https://github.com/cirruslabs/vetu/pull/38/commits/42b08b499c6686fdae1225c4f23809d706c508c6.

Reply:
What do you think about keeping the `globallock.With()` in the outer scope, but only moving the VM locking calls into the `Open()`?

Similarly to how https://github.com/cirruslabs/vetu/commit/42b08b499c6686fdae1225c4f23809d706c508c6 is done.